api/rosetta/data_balance_integration_test.go

```diff
@@ -574,24 +492,30 @@ func TestAPI_BalanceHandlesMalformedRequest(t *testing.T) {
 	)
 
 	tests := []struct {
-		name     string
-		payload  []byte
-		mimeType string
+		name    string
+		payload []byte
+		prepare func(req *http.Request)
 	}{
 		{
-			name:     "wrong field type",
-			payload:  []byte(wrongFieldType),
-			mimeType: echo.MIMEApplicationJSON,
+			name:    "wrong field type",
+			payload: []byte(wrongFieldType),
+			prepare: func(req *http.Request) {
+				req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+			},
 		},
 		{
-			name:     "unclosed bracket",
-			payload:  []byte(unclosedBracket),
-			mimeType: echo.MIMEApplicationJSON,
+			name:    "unclosed bracket",
+			payload: []byte(unclosedBracket),
+			prepare: func(req *http.Request) {
+				req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+			},
 		},
 		{
-			name:     "valid payload with no mime type set",
-			payload:  []byte(validJSON),
-			mimeType: "",
+			name:    "valid payload with no MIME type set",
+			payload: []byte(validJSON),
+			prepare: func(req *http.Request) {
+				req.Header.Set(echo.HeaderContentType, "")
+			},
```

Comment:
It seems like the only thing that changes is the value of `echo.HeaderContentType` and we always do only one operation for the prepare phase?

In that case, it makes more sense IMO to just have a string in the test vector inputs, called smth like `contentType`, and then we always do `req.Header.Set(echo.HeaderContentType, test.contentType)` in the test body, don't you think? 🤔 Or maybe I'm missing something because I only see the changes from this PR.

Reply:
I don't think it's any more complex, though, and it's more flexible. So I'm OK with leaving it.